portfolio/src/main/webapp/script.js

```diff
@@ -151,3 +151,15 @@ function updateButton() {  // eslint-disable-line no-unused-vars
     document.getElementById('comment-button').disabled = false;
   }
 }
+
+function checkLoginStatus() {  // eslint-disable-line no-unused-vars
+  fetch('/login').then((response) => response.text()).then((message) => {
+    if (message === 'Logged In\n') {
```

Comment:
Consider not doing string checks: for example, `LoginServlet` returns an object that's something like 
```
{
  email: XXX, 
  name: XXX,
  isLoggedIn: true,
  loginUrl: XXX
  // other fields
}
```
You can either use a JSON object as per last week, or use a protobuf: https://developers.google.com/protocol-buffers/docs/javatutorial
(this might take a bit of setting up, let me try it and see if i can get a working example)

Reply:
I changed the doGet() function to return a JSON object instead of a string.